overlord/snapstate/storehelpers.go

```diff
@@ -637,6 +637,15 @@ func refreshCandidates(ctx context.Context, st *state.State, names []string, rev
 		return nil, nil, nil, err
 	}
 
+	opts, err = refreshOptions(st, opts)
```

Comment:
this is a bit overkill, I would just change the if to:

```
if opts == nil || !opts.Scheduled { ...
```

perhaps

Reply:
a yet different approach is to remove the use of refreshOptions in the ...Core variant of the function and call it here before calling ...Core